api/v1alpha3/machineset_types.go

```diff
@@ -175,8 +175,8 @@ func (m *MachineSet) Validate() field.ErrorList {
 	if err != nil {
 		errors = append(errors, field.Invalid(fldPath.Child("selector"), m.Spec.Selector, "invalid label selector."))
 	} else {
-		labels := labels.Set(m.Spec.Template.Labels)
-		if !selector.Matches(labels) {
+		lbls := labels.Set(m.Spec.Template.Labels)
```

Comment:
Personal opinion: 
I understand that package name shadowing is not ideal, but I find that forcibly changing variable names to something else makes the code less readable in most cases (e.g cfg is fine for config, but lbls, scrt, vrsn, ptch are kind of cryptic to me).

Reply:
+1

Reply:
so would it be better to drop this PR or try to come up with meaningful names in each place where they're used? An alternative would be to change those imports with generic names where they're introduced, but that's probably not ideal either.

Reply:
+1 to dropping this PR and linter, if a package is shadowed and it becomes an issue an implementer can decide to override the package name or the variable that's shadowing it